app/assets/javascripts/common/reports.js

```diff
@@ -1699,3 +1678,21 @@ const intersectDataVerticalLine = {
   },
 };
 
+// ---
+// Object merging functions
+function combineConfigWithBaseConfig(additionalConfig) {
```

Comment:
I'd say we can even name this method `withBaseConfig`.

Reply:
Done